include/hw/arm/guest-services/socket.h

```diff
@@ -32,6 +32,39 @@
 #include "sys/socket.h"
 #endif
 
+#ifdef __APPLE__
+#define S_SOCKADDR	struct sockaddr
+#define S_SOCKADDR_IN	struct sockaddr_in
+#define SOCKLEN_T	socklen_t
+#else /* __linux__ */
+/* XXX: A lot of this is defined by the standard.
+	But doesn't hurt to ensure Darwin/BSD lengths.
```

Comment:
I get what you're saying, but since `socklen_t` seems to be the same between GNU and Darwin, I'd avoid redefining it - the less specialized types we define here, the better. So let's just use `socklen_t`.